app/components/primer/alpha/dialog.rb

```diff
@@ -0,0 +1,138 @@
+      renders_one :body, lambda { |**system_arguments|
+        deny_tag_argument(**system_arguments)
+        system_arguments[:tag] = :div
+        system_arguments[:classes] = class_names(
+          system_arguments[:classes]
+        )
+        Primer::BaseComponent.new(**system_arguments)
+      }
+
+      # Footer content.
+      #
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :footer, lambda { |**system_arguments|
+        deny_tag_argument(**system_arguments)
+        system_arguments[:tag] = :div
+        system_arguments[:classes] = class_names(
+          "Overlay-footer",
+          "Overlay-footer--alignEnd",
+          "Overlay-footer--divided",
+          system_arguments[:classes]
+        )
+        Primer::BaseComponent.new(**system_arguments)
+      }
+
+      # @example Dialog with Cancel and Submit buttons
+      #   @description
+      #     An ID is provided which enables wiring of the open and close buttons to the dialog.
+      #   @code
+      #     <%= render(Primer::Alpha::Dialog.new(
+      #       title: "Dialog Example",
+      #     )) do |d| %>
+      #       <% d.show_button { "Show Dialog" } %>
+      #       <% d.body do %>
+      #         <p>Some content</p>
+      #       <% end %>
+      #       <% d.footer do %>
+      #         <%= render(Primer::ButtonComponent.new(data: { "close-dialog-id": "my-dialog" })) { "Cancel" } %>
+      #         <%= render(Primer::ButtonComponent.new(scheme: :primary)) { "Submit" } %>
+      #       <% end %>
+      #     <% end %>
+      # @param id [String] The id of the dialog.
+      # @param title [String] The title of the dialog.
+      # @param subtitle [String] The subtitle of the dialog. This will also set the `aria-describedby` attribute.
+      # @param width [Symbol] The width of the dialog. <%= one_of(Primer::Alpha::Dialog::WIDTH_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      # @param show_header_divider
+      # @param show_footer_divider
+      # @paranm body_padding_variant??
+      def initialize(
+        title:,
+        subtitle: nil,
+        width: DEFAULT_WIDTH,
+        id: "dialog-#{(36**3 + rand(36**4)).to_s(36)}",
+        **system_arguments
+      )
+        @system_arguments = deny_tag_argument(**system_arguments)
+
+        @system_arguments[:tag] = "modal-dialog"
+        @system_arguments[:role] = "dialog"
+        @system_arguments[:id] = id.to_s
+        @system_arguments[:aria] = { modal: true }
+        @system_arguments[:classes] = class_names(
+          "Overlay",
+          WIDTH_MAPPINGS[fetch_or_fallback(WIDTH_OPTIONS, width, DEFAULT_WIDTH)],
+          "Overlay--height-auto",
+          "Overlay--motion-scaleFade",
+          system_arguments[:classes]
+        )
+
+        if subtitle.present?
+          @subtitle = subtitle
+          @subtitle_id = "#{id}-description"
+          @system_arguments[:aria].describedby ||= @description
```

Comment:
```suggestion
          @system_arguments[:aria].describedby ||= @subtitle
```

I _think_ this is meant to be `subtitle` now